terraform/environments/staging/main.tf

```diff
@@ -532,6 +532,123 @@ resource "google_project_iam_member" "application" {
   member = "serviceAccount:${each.value}"
 }
 
+# Deploy relays
+
+module "relays" {
+  source     = "../../modules/relay-app"
+  project_id = module.google-cloud-project.project.project_id
+
+  instances = {
```

Comment:
@jamilbk This will cost 379$ monthly but will allow us to test minimal a global deployment and see how traffic flows in such setup.

Reply:
This should hit our credits though correct? I think this cost is reasonable -- do you have any feeling for how this cost might scale?

Reply:
Link to instance specs for @thomaseizinger and @conectado --

https://www.economize.cloud/resources/gcp/pricing/n1/n1-standard-1/

Reply:
@jamilbk Yes this will hit our credits so nothing to worry about.

> Do you have any feeling for how this cost might scale?

There are two cost factors:
1. Instance price. I'm using the smallest instances available (1 cou core and 3.75 Gbps ram) but we might want to scale them depending on how much resources the relay will actually consume. I assume it will be CPU and IO-bound, which means we might want to go for n1-highcpu-2 (2 CPU, 1.8 Gb ram, 10 Gbps) which will end up ~500$. n1 should have 2 Gbps link so such a setup can be more than enough even for our v1 production.
2. Traffic. Can't comment on this one, depends on how many symmetric NATs we will encounter IRL.

Reply:
@thomaseizinger what do you think we will need resource-wise? What it should be bound by?

Reply:
My guess is that we will be primarily IO-bound. The most CPU heavy thing is the cryptography regarding authentication and that is minimal. One of the first things we should look at is increasing UDP buffer sizes. More than 1 CPU could make sense too because each "allocation" (i.e. incoming traffic that needs to be relayed) is running on a separate task (read: green-thread) and those will be run concurrently by the (multi-threaded) runtime.